### DISCUSSION TITLE
More love for TDD

### DISCUSSION THREAD

**Discussion by nalbion** [Pythagora-io/gpt-pilot#66] [Ideas]
What's with all of the hating on unit tests?

> You do not need to list any technologies related to automated tests like Jest, Cypress, Mocha, Selenium, etc

> You do not need to make any automated tests work

**Comment by nalbion** [comment]
And I'd prefer the opposite of this one - give me a Dockerfile, Cloudformation templates, config for Serverless Framework...

> **IMPORTANT**
> Do not tell me anything about setting up the database or anything OS related - only if some dependencies need to be installed

**Comment by zvone187** [comment]
Where did you see the hating on unit tests? They will likely become a bedrock of GPT Pilot once they get implemented.

**Reply by nalbion** [reply]
https://github.com/Pythagora-io/gpt-pilot/blob/bcfefd019d2c12248a999c662cdf072ce0c2d778/pilot/prompts/architecture/technologies.prompt#L32

**Comment by lucas-0liveira** [comment]
It seems like there might be some misunderstanding here. The criticism of unit tests could stem from various perspectives within the field of software development. Some developers might feel that writing and maintaining unit tests can be time-consuming, especially if not done efficiently. Additionally, there could be situations where unit tests are not prioritized due to tight deadlines or a perceived lack of immediate benefits.

However, it's important to note that unit tests play a crucial role in ensuring the reliability and stability of codebases. They help catch bugs early in the development process, promote code modularity, and provide documentation for how functions and components are intended to work. Neglecting unit tests can lead to a higher likelihood of undetected bugs and more challenging debugging processes in the long run.

Ultimately, the perspective on unit tests can vary depending on the specific project, team, and development philosophy in place. While they might not always be favored, they are generally considered an essential part of maintaining high-quality software.

**Comment by MaxAnfilofyev** [comment]
related to https://github.com/Pythagora-io/gpt-pilot/discussions/878
